### ISSUE TITLE
Project does not compile after any change is done

### ISSUE THREAD

**Issue by abgonzalez93** [latitude-dev/latitude#320]
## Steps To Reproduce

I have my project running on Ubuntu.
I do a change on my code (View, queries, whatever).
After saving the file, the project does not detect the new changes.
You have to stop and re-run it again.

### Environment

- Node version (`node -v`): 18.20.2
- npm version (`npm -v`): 10.5.2
- OS: Ubuntu 22.04.3**

## Expected Behavior

Is not it supposed to be reactive?
_(Every change is compiled in real-time)_

**Comment by geclos** [comment]
hey @abgonzalez93 it is indeed supposed to be reactive. 

What Latitude CLI version and Latitude project version are you on?
- latitude cli version: `latitude --version`
- latitude project version: check `latitude.json` at the root of your project you'll see the version there

**Comment by abgonzalez93** [comment]
**latitude cli version: latitude --version**

![imagen](https://github.com/latitude-dev/latitude/assets/39385878/1be80d93-df63-4499-a3b5-eb6cec2258f4)

latitude project version: check latitude.json at the root of your project you'll see the version there

![imagen](https://github.com/latitude-dev/latitude/assets/39385878/2121ec37-bcd6-4f0b-a00c-496e2667b447)

**Comment by geclos** [comment]
@abgonzalez93 I'm not able to reproduce. Could you share a repo with the reproducible bug? Also try updating to latest CLI version, there's a bugfix related to CLI syncing that might be affecting you. `npm update -g @latitude-data/cli`

**Comment by abgonzalez93** [comment]
**CLI is up to date.**

Whenever I save a file after a change is done, I get those errors on console (_but it's supposed to be a favicon missing error_)
![imagen](https://github.com/latitude-dev/latitude/assets/39385878/b5675343-ce43-4486-8509-2bf916ef1421)

**I've uploaded Netflix test project to my GitHub repo.**
(.latitude folder is missing. Github doesnt upload it. I suppose doing **_latitude setup_** command you download it)
https://github.com/abgonzalez93/latitude-syncing-error.git

**Comment by geclos** [comment]
hey @abgonzalez93 I'm not able to reproduce your issue with the repo you shared. My settings just so we are in sync:
- ubuntu 22.04
- latitude project v 0.11.0
- latitude CLI version 0.8.2
- node version 18

Some general recommendations that I've solving issues in other projects running in ubuntu:
- run `apt install -y build-essential xdg-utils`. Usually these come preinstalled with ubuntu but might not be your case
- increase the inotify watch limit. [Guide here](https://dev.to/stephencweiss/running-vscode-on-linux-hitting-my-watcher-limit-and-what-to-do-about-it-52i0).
- If you are using wsl, it has some [known issues ](https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc) with file watching that might be interfering with latitude's watchers. But this is just an hypothesis, since I don't have a windows machine to test with.

Let me know if any of this helps

**Comment by abgonzalez93** [comment]
I'm actually using WSL, so that might be a possible cause.
Anyway, I'm gonna try to reinstall build-essentials and search for information about WSL issues.
Ill let you know as soon as I know what's happening

**Comment by abgonzalez93** [comment]
I got to solve it.
I'm currently using WSL on my Windows 11 computer.
I was connecting to WSL files vía VS Code (but using a directory URL) and somehow, as you said, that interfered with latitude's watchers.
I downloaded and configured WSL extension. Accessed to the project and now, running Ubuntu on VS Code terminal, changes are syncing perfectly.

https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl

Hope this helps to others.
